protocol/x/clob/keeper/deleveraging.go

```diff
@@ -162,21 +162,26 @@ func (k Keeper) OffsetSubaccountPerpetualPosition(
 	)
 
 	numSubaccountsIterated := 0
+	numSubaccountsWithNonOverlappingBankruptcyPrices := 0
+	numSubaccountsWithNoOpenPositionOnOppositeSide := 0
 	deltaQuantumsRemaining = new(big.Int).Set(deltaQuantumsTotal)
 	fills = make([]types.MatchPerpetualDeleveraging_Fill, 0)
 
-	s := rand.NewSource(k.blockTimeKeeper.GetPreviousBlockInfo(ctx).Timestamp.Unix())
-	rand := rand.New(s)
-
 	k.subaccountsKeeper.ForEachSubaccountRandomStart(
 		ctx,
 		func(offsettingSubaccount satypes.Subaccount) (finished bool) {
+			// Iterate at most `MaxDeleveragingSubaccountsToIterate` subaccounts.
+			if numSubaccountsIterated >= int(k.MaxDeleveragingSubaccountsToIterate) {
+				return true
+			}
```

Comment:
Question for my understanding: why not put this at the end of the loop?

Won't this logic technically read `MaxDeleveragingSubaccountsToIterate + 1` subaccounts from state because this finishing condition is at the beginning of the loop (a subaccount is read from state then not used)?

Reply:
putting at the end of loop will technically not be correct when Max = 0. reading an extra subaccount is not really a big deal in my opinion.